dev/ci/README-developers.md

```diff
@@ -49,9 +49,40 @@ for more details.
 
 ### Breaking changes
 
-When your PR breaks an external project we test in our CI, you must
-prepare a patch (or ask someone—possibly the project author—to
-prepare a patch) to fix the project. There is experimental support for
+When your PR breaks external projects we test in our CI, you must:
+
+1. Assess the breakage.
+   * If it is a bug your PR introduces, it should be fixed.
+   * Otherwise, you must assess the impact of the breakage on a few CI
+     entries at the bottom of its hierarchy, typically mathcomp,
+     stdlib and stdpp.
+2. Some breakages are legitimate, for instance to remove something
+   already deprecated for long. Less obvious cases are to be
+   determined with with the PR assignee and reviewers, or discussed
+   during a [weekly Call](https://github.com/coq/coq/wiki/Coq-Calls)
+   when doubts remain.
+3. For legitimate breakages, you must then write porting instructions
+   (typically the PR changelog), based on your previous assesment.
+4. The PR assignee will finally ask the project maintainers to prepare
+   a patch. For plugins, you must prepare yourself an overlay (you can
+   ask the CI project maintainer for help). Ultimately, for
+   non-plugin CI projects, the responsibility of preparing a patch
+   falls on the project maintainer, but PR writers are encouraged to
+   be helpful and may prepare patches themselves to facilitate a
+   smooth process of adaptation. To ask project maintainers to adapt
```

Comment:
This is where we could mention the possibility of providing a porting scripts when many lines need to be updated (as per Jason's suggestion).

Reply:
I rather put it into the previous item (mentioning porting instructions).